### PR TITLE
Add hierarchical template handling

### PR DESCRIPTION
**Pull request by T0jan**
* adds an option to define parent category parameters in supplier_parameters.yaml
* adds support to define a default image for a category in supplier_parameters.yaml

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -194,6 +194,10 @@ CATEGORY_NAME:
     - SUPPLIER_2_PARAMETER_NAME_1
 ```
 
+It is also possible to cross reference the mappings of different categories. To define one or multiple parent categories a parameter named `parent` can be added where the items then are the desired parent categories. If a parameter name is present in both parent and child, the childs definition will override the parent.
+
+A template image for an category can be set by using the `image` parameter. The sole item in this parameter must the filename of an already existing part image on the the InvenTree server.
```
Comment (eeintech) [inline README.md:199]: I don't understand this concept, can you explain?

File: kintree/database/inventree_interface.py [hunk 2]
```diff
@@ -286,7 +286,11 @@ def translate_form_to_inventree(part_info: dict, category_tree: list, is_custom=
             for supplier_param, inventree_param in parameter_map.items():
                 # Some parameters may not be mapped
                 if inventree_param not in inventree_part['parameters'].keys():
-                    if supplier_param != 'Manufacturer Part Number':
+                    if supplier_param == 'Manufacturer Part Number':
+                        inventree_part['parameters'][inventree_param] = part_info['manufacturer_part_number']
+                    elif inventree_param == 'image':
+                        inventree_part['existing_image'] = supplier_param
```
[the remark this answers is not in the document]
Comment (T0jan) [reply kintree/database/inventree_interface.py:292]: @eeintech not just an naming update, its an new option added to the API: https://github.com/inventree/InvenTree/pull/6021

with it you don't have to upload the same picture multiple times if you reuse it, instead you can just reference the picture of another part.

File: README.md (hunk 1 above)
Comment (T0jan) [reply README.md:199]: @eeintech imagine you have parameters which are used for all components, something like `Operating Temperature` or `MPN`. With my changes you can define a `base` category, mapping these two (and others) in the `supplier_parameters.yaml`. All other categories then don't need to map these parameters again, instead just map `base` as their parent and then use the `base` mappings as well as their own. 

An example where I will heavily use it is for different resistor sizes: In our InvenTree we differentiated our Resistors by their size, so there is a Resistors/0603 category, a /0402 and so on. Each of these categories use a predefined image to visually differentiate them. With the `existing_image` functionality I would have to define all the different categories in the `suppliers_parameters.yaml` with all mappings again and again and again. Instead I now just can say that the `Resistors/0603` category uses the `Resistors` mapping and only adds the definition of the image which will be used.

File: kintree/database/inventree_interface.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (eeintech) [reply kintree/database/inventree_interface.py:292]: I see, makes sense. I haven't been following the latest development, thanks for keeping this up-to-date!

File: README.md (hunk 1 above)
Comment (eeintech) [reply README.md:199]: The reuse of parameters across categories is a really great addition, the one part I was missing is the `image` parameter, I was just confused of its usage. So now let's say a part found via supplier API does not return an image, the default `image` for its category will be used instead, is that right?

Comment (eeintech) [reply README.md:199]: @T0jan Understood, I was a little concerned that existing image will make an entire database _look_ the same if it is preferred to supplier image but I guess it is up to the user to decide to specify an image path in the config file so it has to be intentional for that to happen.

I'll try to test and merge it next week 👍

**Pull request merged**